### PR TITLE
Emit WorkUnitsCreated Count Event for MR deployed jobs. 

### PR DESCRIPTION
**Pull request by krishraman**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-766] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-766

### Description
- [ x] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Tested on Yugioh by emitted the events to kafka and verifying using kafka-tool

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by krishraman** [comment]
Hi @htran1 @zxcware Can you please review this PL? Thanks

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java [hunk 1]
```diff
@@ -262,6 +265,10 @@ protected void runWorkUnits(List<WorkUnit> workUnits) throws Exception {
     JobState jobState = this.jobContext.getJobState();
 
     try {
+      CountEventBuilder countEventBuilder = new CountEventBuilder(JobEvent.WORK_UNITS_CREATED, workUnits.size());
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java]: Why not emit WorkUnitsCreated for Local and Helix jobs as well?

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/local/LocalJobLauncher.java [hunk 2]
```diff
@@ -130,20 +134,30 @@ protected void runWorkUnitStream(WorkUnitStream workUnitStream) throws Exception
     Iterator<WorkUnit> workUnitIterator = workUnitStream.getWorkUnits();
     if (!workUnitIterator.hasNext()) {
       LOG.warn("No work units to run");
+      CountEventBuilder countEventBuilder = new CountEventBuilder(JobEvent.WORK_UNITS_EMPTY, 0);
+      this.eventSubmitter.submit(countEventBuilder);
       return;
     }
 
+    this.workUnitCount = 0;
+    while (workUnitIterator.hasNext()) {
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/local/LocalJobLauncher.java]: Can you replace the while loop with a for loop? e.g. for (i=workUnitIerator; i.next();) {...}

Further the 2 countEventBuilders can be combined into a single statement at the end of the for loop:
e.g. CountEventBuilder countEventBuilder = (workUnitCount=0)? new CountEventBuilder(JobEvent.WORK_UNITS_EMPTY, 0) : new CountEventBuilder(JobEvent.WORK_UNITS_CREATED, this.workUnitCount);

**Comment by sv2000** [comment]
@krishraman Can you check travis failure?

**Comment by krishraman** [comment]
Iterating the workunit Stream in the localJoblauncher is not possible because the same iterator is returned and then trying to get a flattened Wu's returns empty. Made a change to return flattened Wu's instead.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java (hunk 1 above)
Comment (sv2000) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java]: So we are no longer emitting workUnitsCreated timer for LocalJobLauncher?

Comment (krishh85) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java]: We are workUnitsCreatedTimer events as well. That has not changed for local job launcher.

**Comment by krishh85** [comment]
For LocalJob Launcher we can use the taskCount in taskSubmitted Event which is a representation of flattened workunits created.

**Pull request closed without merging**